massa-protocol-exports/src/protocol_controller.rs

```diff
@@ -39,18 +43,16 @@ pub enum ProtocolEvent {
         /// The header
         header: WrappedHeader,
     },
-    /// Ask for a list of blocks from consensus.
-    GetBlocks(Vec<BlockId>),
 }
 /// Possible types of pool events that can happen.
-#[derive(Debug, Serialize)]
+#[derive(Debug)]
 pub enum ProtocolPoolEvent {
     /// Operations were received
     ReceivedOperations {
         /// the operations
         operations: Map<OperationId, WrappedOperation>,
-        /// whether or not to propagate operations
-        propagate: bool,
+        /// whether or not to signal the end of processing the batch.
+        done_signal: Option<oneshot::Sender<()>>,
```

Comment:
I don't understand why this change is made ?

Reply:
To make protocol block on pool while it processes the operation, and adds to them storage, so that when protocol then re-creates the block and sends it to consensus, we are sure the operation are in storage. Could be removed perhaps with the latest changes to storage...